sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CommitLog.scala

```diff
@@ -51,6 +51,8 @@ class CommitLog(sparkSession: SparkSession, path: String)
 
   import CommitLog._
 
+  private val VERSION = SQLConf.get.stateStoreCheckpointFormatVersion
```

Comment:
Why is it moved here?

Reply:
As Burak said here, https://github.com/apache/spark/pull/49275#discussion_r1906535722

the conf can be changed per session, and in the test we need to change the conf during test.